### DISCUSSION TITLE
Ability to employ manufactures' models for components

### DISCUSSION THREAD

**Discussion by csrabak** [ra3xdh/qucs_s#257] [Ideas]
I tried to find about this in the present documentation, and except due a terminology mishap of mine, I did not find anything.

Since the qucs-s approach to employ engines which in their guts use Spice as a _lingua franca_ for circuit simulation, I wonder how easy would it be to implemente and/or document the use of spice models of different manufacturers.

Even some more prosaic models could be useful as the ones exported by [ Coil 32 ](https://coil32.net), for example, in the evaluation of circuits where the anoying imperfections of the coils make some responses lacking, etc.

Perhaps, due the use of more than a single engine (namely NG Spice, Xyce, and SpiceOpus) the instructions may need to be deferred to the specific engine?

However, I think having qucs-s as an unifying plattform for these chores would be a plus for this project!

--
Cesar Rabak

**Comment by ra3xdh** [comment]
> I tried to find about this in the present documentation, and except due a terminology mishap of mine, I did not find anything.

This manual section explains how to attach a SPICE opamp model to Qucs-S https://qucs-s-help.readthedocs.io/en/latest/SubLib.html#using-manufacturers-component-data-libraries 

There exists two devices that may be used for this purpose:

* File components->SPICE netlist
* File components->SpiceLibComp

It's need to kep in mind that SPICE models don't have an unified syntax. And for example LTSpice models may contain some unique syntax constructions that need to be rewritten to use this model with Ngspice. This task cannot be performed automatically.

**Reply by csrabak** [reply]
Yes I do understand in full these variations.

However, this comment of yours gives rise to another "opportunity": NG Spice has the feature of allowing to choose some compatibility option[s] so to swallow without need to further processing (i.e. IT does the preprocessing/conversion) to some more _popular_ syntaxes, I'll try to look for in the docs if in present qucs-s it is possible to set this options via the variable ```ngbehavior```¹.

Thanks

[1] The full set of available options is in Table 16.2 of present NG Spice manual.

**Comment by csrabak** [comment]
I'm still learning where best place to post some so small observations, and if convenient, pls move this from this thread.

In the above indicated link, in third paragraph of **Libraries Blacklisting** the directive in the star phrase has been written as ```*.balcklist````, whereas I assume the correct would have been _blacklist_.

HTH

**Comment by ra3xdh** [comment]
> I'll try to look for in the docs if in present qucs-s it is possible to set this options via the variable 

Qucs-S supports `spiceinit` scripts. And it's possible to define the compatibility options using the spiceinit. See here for more information: https://github.com/ra3xdh/qucs_s/pull/26 This feature still be not documented.
